src/SimpleJson/SimpleJson.cs

```diff
@@ -1412,6 +1415,21 @@ public virtual object DeserializeObject(object value, Type type)
                             dict.Add(kvp.Key, DeserializeObject(kvp.Value, valueType));
 
                         obj = dict;
+
+#if SIMPLE_JSON_READONLY_COLLECTIONS
+                        // Wrap dictionary in a ReadOnlyDictionary<,>
+                        var genericTypeDefinition = type.GetGenericTypeDefinition();
+                        if (genericTypeDefinition == typeof(IReadOnlyDictionary<,>) ||
+                            genericTypeDefinition == typeof(ReadOnlyDictionary<,>))
+                        {
+                            var ctorType = typeof(IDictionary<,>).MakeGenericType(keyType, valueType);
+                            var genericReadonlyType = typeof(ReadOnlyDictionary<,>).MakeGenericType(keyType, valueType);
+                            var ctor = ReflectionUtils.GetContructor(genericReadonlyType, new Type[] { ctorType });

```

Comment:
We can actually migrate the tests for to Net45 instead of adding a new csproj. But make sure to update the `jakefile` specially this line of code. https://github.com/facebook-csharp-sdk/simple-json/blob/f739f0b2924bcfa33204e86b8b73aa98be68abac/jakefile.js#L231-L255. Also make sure `jake clean` and `jake` works. You can learn about how to install/build from jake at http://facebooksdk.net/docs/build/